storage/innobase/os/os0file.cc

```diff
@@ -1133,13 +1131,11 @@ static ATTRIBUTE_COLD bool os_file_log_maybe_unbuffered(const struct stat &st)
   if (s > 4096 || s < 64 || !ut_is_2pow(s))
     return false;
   log_sys.set_block_size(uint32_t(s));
-# else
-  constexpr unsigned long s= 4096;
-# endif
 
   return !(st.st_size & (s - 1));
 }
-#endif
+#endif /* __linux__ */
+#endif /* O_DIRECT */
```

Comment:
I would suggest to add the missing indentation, either instead or in addition to adding those comments:
```c
# endif /* __linux__ */
#endif /* O_DIRECT */
```